src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2548,16 +2536,10 @@ Status Session::receive_unbound_message()
 }
 
 
-Status Session::receive_query_error_message(int error_code, std::string_view message, int64_t query_version)
+void Session::receive_query_error_message(int error_code, std::string_view message, int64_t query_version)
 {
     logger.info("Received QUERY_ERROR \"%1\" (error_code=%2, query_version=%3)", message, error_code, query_version);
-    // Ignore the message if the deactivation process has been initiated,
-    // because in that case, the associated Realm and SessionWrapper must
-    // not be accessed any longer.
-    if (m_state == Active) {
```

Comment:
Checking for m_state == Active was done several time along the way so I eliminated all but one.